sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client_helpers.py

```diff
@@ -0,0 +1,82 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+# pylint: disable=docstring-keyword-should-match-keyword-only
+
+from typing import (
+    Any, Dict, Optional, Tuple, Union,
+    TYPE_CHECKING
+)
+from urllib.parse import quote, unquote, urlparse
+
+from ._deserialize import deserialize_permission_key
+from ._generated.models import SharePermission
+from ._shared.base_client import parse_query
+
+if TYPE_CHECKING:
+    from urllib.parse import ParseResult
+
+
+def _parse_url(account_url: str, share_name: str) -> "ParseResult":
+    try:
+        if not account_url.lower().startswith('http'):
+            account_url = "https://" + account_url
+    except AttributeError as exc:
+        raise ValueError("Account URL must be a string.") from exc
+    parsed_url = urlparse(account_url.rstrip('/'))
+    if not share_name:
+        raise ValueError("Please specify a share name.")
+    if not parsed_url.netloc:
+        raise ValueError(f"Invalid URL: {account_url}")
+    return parsed_url
+
+
+def _format_url(scheme: str, hostname: str, share_name: str, query_str: str) -> str:
+    if isinstance(share_name, str):
+        share_name = share_name.encode('UTF-8')
+    return f"{scheme}://{hostname}/{quote(share_name)}{query_str}"
+
+
+def _from_share_url(share_url: str, snapshot: Optional[Union[str, Dict[str, Any]]]) -> Tuple[str, str, str]:
+    try:
+        if not share_url.lower().startswith('http'):
+            share_url = "https://" + share_url
+    except AttributeError as exc:
+        raise ValueError("Share URL must be a string.") from exc
+    parsed_url = urlparse(share_url.rstrip('/'))
+    if not (parsed_url.path and parsed_url.netloc):
+        raise ValueError(f"Invalid URL: {share_url}")
+
+    share_path = parsed_url.path.lstrip('/').split('/')
+    account_path = ""
+    if len(share_path) > 1:
+        account_path = "/" + "/".join(share_path[:-1])
+    account_url = f"{parsed_url.scheme}://{parsed_url.netloc.rstrip('/')}{account_path}?{parsed_url.query}"
+
+    share_name = unquote(share_path[-1])
+    path_snapshot, _ = parse_query(parsed_url.query)
+    if snapshot:
+        try:
+            path_snapshot = snapshot.snapshot  # type: ignore
+        except AttributeError:
+            try:
+                path_snapshot = snapshot['snapshot']  # type: ignore
+            except TypeError:
+                path_snapshot = snapshot
```

Comment:
The above 7 lines should be refactored is use `isinstance`, etc. and not have type ignores.

This can be handled in the separate actual typing PR but I just don't want it to get missed since the ignores will make it not show up in the MyPy run :)